### ISSUE TITLE
Bad Allocation error when browsing a table at QGIS 3.28.8

### ISSUE THREAD

**Issue by Mmito24** [qgis/QGIS#55238]
### What is the bug or the crash?

When trying to open the attributes of a table (for browsing it),  QGIS sends the exception message "Bad Allocation" before finishing reading all the features and then closes the table (so the table is never showed).
It seems this is a generic message related to memory problems, so I split my layer in two, thus I have now two layers with 2 tables associated, with half of the values each.
It worked fine for a while, but then one of the halfs started showing the same "Bad Allocation" exception. The wierd thing is that the table which is showing the message is the one with less points (I think they are 22 million points against 24 million points in the other half).
I tried remaking (or redoing?) the table with the "remake fields" tool (sorry if the name of the tool is slighlty different, my QGIS version is not in english), but I got the same problem on the new table.
I need to populate some of the fields in those tables frequently, but I can't do that if the attributes of table does not open.

BTW, I'm running QGIS on a server with 2 Intel Xeon Processors, Hard disk drives and 128 Gb RAM, so I don't think it would be due to a lack on resources.

### Steps to reproduce the issue

1. With open data sources, open the vector layer "bins_25m_I2020w84Sur.gpkg"
2. Right click over the layer "bins_25m_I2020w84Sur.gpkg" and select "Open table attributes"
3. After loading several features, QGIS show exeption "Bad Allocation"

### Versions

Versión de QGIS
3.28.8-Firenze
Revisión del código de QGIS
5ac45272b5
Versión Qt
5.15.3
Versión de Python
3.9.5
Versión de GDAL/OGR
3.7.0
Versión de PROJ
9.2.1
Versión del registro de base de datos EPSG
v10.088 (2023-05-13)
Versión de GEOS
3.11.2-CAPI-1.17.2
Versión de SQLite
3.41.1
Versión de PDAL
2.5.3
Versión del cliente de PostgreSQL
unknown
Versión de SpatiaLite
5.0.1
Versión de QWT
6.1.6
Versión de QScintilla2
2.13.1
Versión del SO
Windows Server 2012 R2 Version 6.3 (Build 9600)

Complementos activos de Python
dzetsaka
3.70
quick_map_services
0.19.33
SemiAutomaticClassificationPlugin
8.1.3
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99
sagaprovider
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

<img width="953" alt="Bad Allocation when browsing a table" src="https://github.com/qgis/QGIS/assets/150381996/6a375c2e-ce4e-46a5-8631-4e230bb3b44a">

**Comment by alexbruy** [comment]
Please attach dataset which causes crash. Also please try with the new profile without any 3rd party plugins installed.

**Comment by Mmito24** [comment]
I am sorry, it seems that the dataset can´t be attached. Apparently size is too big. I am attempting to upload 669 MB (already compressed). Can I upload it some where else without this limitation?
Also I don't think I have 3rd party plugins, since I have not installed any plugin by myself (unless they are already installed in the distribution).
Thank you for your follow up.

**Comment by alexbruy** [comment]
@Mmito24 you can upload it to any file sharing service and post the link to the data here.

Regarding 3rd party plugins, in your QGIS details I see 3 of them hence the question.

**Comment by alexbruy** [comment]
I can't reproduce bad allocation error with master and 3.28.13 on a PC with just 16 Gb or RAM. Opening attribute table takes time, but there is no crash or errors afterwards. Have you checked you RAM for errors? Are you working via remote desktop or similar protocols?

Current implementation of the attribute table is not working well with huge tables, there are some plans to improve this. Meanwhile you can use other ways to edit features: field calculator, feature forms, etc.

**Comment by Mmito24** [comment]
Hi Alex,

Have you checked you RAM for errors? No, I haven't (and sorry, but I don't have any idea on how to do that)
Are you working via remote desktop or similar protocols? Yes, I am working via remote desktop to connect to the server.

And I made a test to open the table in a new profile apparently without additional plugins and the problem didn't appear. I'm going to try with the complete database which is bigger and hope it will open fine too. I'll let you know.

;-)

**Comment by GISAPP-Enterprise** [comment]
FYI.. may help u troubleshoot your issue..
I had seen same error while opening attribute table in 3.28. But I realized it was related to the field calculator expressions I had set in form designer in that gpkg's properties. The field calculator expressions I had set there were referring to some tables and attributes which were not present in the new project file. Once I removed it the attribute table opened again.
